app/models/katello/candlepin/repository_mapper.rb

```diff
@@ -13,7 +13,10 @@ def find_repository
         root = ::Katello::RootRepository.where(product_id: product.id,
                                     content_id: content.cp_content_id,
                                     minor: minor,
-                                    arch: arch).first
+                                    arch: arch).or(
+               ::Katello::RootRepository.where(product_id: product.id,
+                                               content_id: content.cp_content_id,
+                                               minor: minor)).first
```

Comment:
this seemed to cause a bug, if i go to the RH repos page and search for 'label = rhel-6-server-rpms'  
Then enable the 'x86_64 6Server' repo, it won't let me enable the 'i386 6Server' repo.  I think because its not finding the arch i specified (i386), so then it excludes the arch in the search and finds the x86_64 one, os it thinks its already enabled.  Should we search for the arch passed in or nil?

Reply:
something like this (untested)
```
::Katello::RootRepository.where(product_id: product.id,
                                    content_id: content.cp_content_id,
                                    minor: minor,
                                    arch: [arch, nil]).first
```

Reply:
Oh, good catch. Yeah, I think searching for `nil` in the second search would work. I'm not sure I'll check :).

Reply:
so it looks the issue with that would be the arch is set to "noarch" by default, so it arch would never be nil.

Fixing the duplicate repository issue might be a tricky. I have to dig around again.